File: tasks/implementations/python/tasks/linear_regression.py
import csv
from pathlib import Path

def run(size: str, fixtures_root: str) -> None:
    path = Path(fixtures_root) / "generated" / "linear_regression" / f"{size}.csv"
    n = 0.0
    sum_x = sum_y = sum_xy = sum_x2 = 0.0
    with path.open("r", encoding="utf-8") as handle:
        reader = csv.DictReader(handle)
        for row in reader:
            x = float(row["x"])
            y = float(row["y"])
            n += 1.0
            sum_x += x
            sum_y += y
            sum_xy += x * y
            sum_x2 += x * x
    slope = (n * sum_xy - sum_x * sum_y) / (n * sum_x2 - sum_x * sum_x)
    intercept = (sum_y - slope * sum_x) / n
    checksum = round(slope * 1_000_000) + round(intercept * 1_000_000)
    print(checksum)

File: tasks/implementations/go/tasks/linear_regression.go
package main

import (
    "encoding/csv"
    "fmt"
    "math"
    "os"
    "path/filepath"
    "strconv"
)

func RunLinearRegression(size string, fixturesRoot string) {
    file, _ := os.Open(filepath.Join(fixturesRoot, "generated", "linear_regression", size+".csv"))
    defer file.Close()
    rows, _ := csv.NewReader(file).ReadAll()
    n := 0.0
    sumX, sumY, sumXY, sumX2 := 0.0, 0.0, 0.0, 0.0
    for _, row := range rows[1:] {
        x, _ := strconv.ParseFloat(row[0], 64)
        y, _ := strconv.ParseFloat(row[1], 64)
        n += 1.0
        sumX += x; sumY += y; sumXY += x * y; sumX2 += x * x
    }
    slope := (n*sumXY - sumX*sumY) / (n*sumX2 - sumX*sumX)
    intercept := (sumY - slope*sumX) / n
    checksum := math.Round(slope*1000000) + math.Round(intercept*1000000)
    fmt.Println(int64(checksum))
}
